patches/minecraft/net/minecraft/world/gen/layer/MixRiverLayer.java.patch

```diff
@@ -0,0 +1,14 @@
+--- a/net/minecraft/world/gen/layer/MixRiverLayer.java
++++ b/net/minecraft/world/gen/layer/MixRiverLayer.java
+@@ -22,7 +22,10 @@
+       if (LayerUtil.func_202827_a(i)) {
+          return i;
+       } else if (j == field_202725_h) {
+-         if (i == field_202721_d) {
++    	 net.minecraft.world.biome.Biome modded_river_biome = net.minecraftforge.common.BiomeManager.getRiver(Registry.field_212624_m.func_148745_a(i));
++         if (modded_river_biome != null) {
++        	return Registry.field_212624_m.func_148757_b(modded_river_biome);
++         } else if (i == field_202721_d) {
```

Comment:
1. That's not how you name variables in Java.
2. Don't patch the if into an elseif. Your if already returns, so you don't need an else-if. That makes this a 2-line patch (`if (moddedRiverBiome != null) return ...`

Reply:
Thanks for pointing that out! Will change that

Reply:
done